src/storage.rs

```diff
@@ -308,6 +317,28 @@ where
             _ => panic!("Storage should already have been stopped!"),
         }
     }
+
+    /// Removes elements up to the given [`idx`] from storage.
+    pub fn garbage_collect(&mut self, idx: u64) {
+        match self.sequence {
+            PaxosSequence::Active(ref mut s) => {
+                s.garbage_collect(idx);
+                self.paxos_state.set_gc_idx(idx);
+            }
+            PaxosSequence::Stopped(_) => {} // todo what to do when paxos is stopped?
+            _ => panic!("Got unexpected intermediate PaxosSequence::None in stopped()"),
+        }
+    }
+
+    /// Sets the garbage collector index in storage.
+    pub fn set_gc_idx(&mut self, index: u64) {
```

Comment:
Remove this function. The storage should always perform the gc and then set the gc_idx internally. A user should not be able to only set the gc_idx without actually garbage collecting.